### DISCUSSION TITLE
How to use Timer

### DISCUSSION THREAD

**Discussion by dhunjoshi** [flet-dev/flet#589] [General]
I Want to increment value of a text in time interval. How can i use timer?

**Comment by FeodorFitsner** [comment]
Sure, there is an example at the second part of this article: https://flet.dev/docs/guides/python/user-controls

**Reply by dhunjoshi** [reply]
Thanks for your quick response. I was using timer for showing webcam image. But its flicker on every frame , is it because of page update? any example for webcam?

**Reply by dhunjoshi** [reply]
import flet as ft
import time, threading
import cv2
import base64
from io import BytesIO
from PIL import Image


class Countdown(ft.UserControl):
    
    def __init__(self, seconds):
        super().__init__()
        self.vid = cv2.VideoCapture(0)
        self.counter = ft.Text(value="0000",expand=1,
                        text_align="center",height=30,)
        
        self.image = ft.Image(src_base64=self.get_base_64_image())

        self.c = ft.Container(
            expand=1,
            height=40,
            bgcolor=ft.colors.AMBER,
            border_radius=5,
            padding=10,
            content=self.counter,            

        )
        
        self.reset_btn  = ft.ElevatedButton(text="Reset", width=120,
        on_click=self.reset_timer)
        self.stop_btn  = ft.ElevatedButton(text="Stop", width=120, on_click=self.stop_timer)
        self.stop = False

        self.seconds = seconds
    
    def get_base_64_image(self):
        ret, frame = self.vid.read()
        if ret==False:
            frame = cv2.imread('assets/images/logo.jpeg')
        _, im_arr = cv2.imencode('.jpg', frame)  # im_arr: image in Numpy one-dim array format.
        im_bytes = im_arr.tobytes()
        im_b64 = base64.b64encode(im_bytes)
        im_b64_str = (str(im_b64).replace("b'","").replace("'",""))
        return im_b64_str

    def reset_timer(self, e):
        print("Reset")
        
        if self.stop :
            self.reset_btn.disabled=True
            self.stop=False
            self.seconds=1200
            self.update_timer()

    def stop_timer(self, e):
        print("Stop")
        self.stop=True
        self.reset_btn.disabled = not self.stop
        self.update()


    def did_mount(self):
        self.running = True
        self.th = threading.Thread(target=self.update_timer, args=(), daemon=True)
        self.th.start()
        print("did_mount")

    def will_unmount(self):
        self.running = False
        print("Unmount")

    def update_timer(self):
        while not self.stop:
            mins, secs = divmod(self.seconds, 60)
            self.counter.value = "{:04d}:{:02d}".format(mins, secs)            
            time.sleep(0.001)
            self.seconds -= 1
            self.image.src_base64=self.get_base_64_image()
            self.update()

    def build(self):
        self.reset_btn.disabled=True
        self.countdown = ft.Column(
            controls=[
                ft.Row(
                        controls=[
                            self.c,
                            self.reset_btn,
                            self.stop_btn,
                        ],
                    ),
                  self.image,   
                ],
            )   
        
       
        return self.countdown

def main(page: ft.Page):    
    c1 = Countdown(1200)    
    
    page.add(c1)

ft.app(target=main,
assets_dir="assets")

**Reply by FeodorFitsner** [reply]
Set `gapless_playback=True` to avoid flickering: https://flet.dev/docs/controls/image#gapless_playback
